### ISSUE TITLE
Wrong warning for duplicate-dependency-class

### ISSUE THREAD

**Issue by krulls** [nebula-plugins/gradle-lint-plugin#52]
Have a simple Java project.
Use following dependencies
```
	compile 'org.hornetq:hornetq-core-client:2.4.5.Final'
	compile 'org.hornetq:hornetq-commons:2.4.5.Final'
```
Getting following warning:
```
warning   duplicate-dependency-class         org.hornetq:hornetq-commons:2.4.5.Final in configuration 'compile' has 2 classes duplicated by org.hornetq:hornetq-core-client:2.4.5.Final
```

If I inspect the libraries there are no duplicate classes.

**Comment by jkschneider** [comment]
This was super interesting. Apparently wherever there is a field defined on an interface, a class is generated in every package that *refers* to it. So for the below interface, every hornetq package that refers to `LOGGER` contains a class named **org/hornetq/utils/HornetQUtilLogger_$logger**.

```java
public interface HornetQUtilLogger extends BasicLogger {
   HornetQUtilLogger LOGGER = Logger.getMessageLogger(...);
}
```
